awxkit/awxkit/cli/options.py

```diff
@@ -178,6 +178,9 @@ def json_or_yaml(v):
                     except Exception:
                         raise argparse.ArgumentTypeError("{} is not valid JSON or YAML".format(v))
 
+                if not isinstance(parsed, dict):
+                    raise argparse.ArgumentTypeError("{} is not valid JSON or YAML".format(v))
```

Comment:
It might be nice to add some verbiage saying that it parsed, but isn't a dictionary. It's good to raise this error, but the next steps for the user are very unclear. It's hard enough for ourselves when we're already used to these quirks.